### PR TITLE
Ontology updates

### PR DESCRIPTION
**Pull request by paluchasz**
* New ontology updates
* Some fixes to KRT
* Fix and tests added for CombinatorialSynonymGenerator
See individual commits for details

### CODE REVIEW

**Comment by paluchasz** [comment]
integration tests passed https://github.com/AstraZeneca/KAZU/actions/runs/11369065257

File: kazu/conf/SynonymGeneration/default.yaml [hunk 1]
```diff
@@ -58,37 +58,37 @@ meddra:
       include_greek: true
     - _target_: kazu.ontology_preprocessing.synonym_generation.TokenListReplacementGenerator
       token_lists_to_consider:
-        -  - anomalous
```
Comment (RichJackson) [inline kazu/conf/SynonymGeneration/default.yaml:61]: not sure whats changed here? is this black formatting?

Comment (RichJackson) [reply kazu/conf/SynonymGeneration/default.yaml:61]: ah I see it's the indent level

File: kazu/ontology_preprocessing/downloads.py [hunk 2]
```diff
@@ -224,7 +224,6 @@ def __init__(self, open_targets_version: str, open_targets_dataset_name: str):
         self.open_targets_version = open_targets_version
 
     @staticmethod
-    @functools.cache
     def _cached_wget(full_url: str, cwd: Path) -> None:
```
Comment (RichJackson) [inline kazu/ontology_preprocessing/downloads.py:227]: should we rename/remove this now?

**Review by RichJackson — CHANGES_REQUESTED** [review]
Just one suggestion to rename the cached method (since it's no longer cached!)

File: kazu/conf/SynonymGeneration/default.yaml (hunk 1 above)
Comment (paluchasz) [reply kazu/conf/SynonymGeneration/default.yaml:61]: yeah sorry when I save in VSCode it has some setting which changes it and produces unneeded diff. It is probably the YAML extension I have

File: kazu/ontology_preprocessing/downloads.py (hunk 2 above)
Comment (paluchasz) [reply kazu/ontology_preprocessing/downloads.py:227]: good spot

**Review by RichJackson — APPROVED** [review]

**Pull request merged**